src/actions/connectionKeyPairActions.js

```diff
@@ -257,6 +257,18 @@ export const updateConnectionKeyPairs = (
     const { currentConnectionsCount, oldConnectionsCount, newReceivedConnectonsCount } = numberOfConnections;
     const totalConnections = currentConnectionsCount + oldConnectionsCount + newReceivedConnectonsCount;
 
+    if (generateKeys && totalConnections === 0 && connectionKeyPairs.length === 0 && lastConnectionKeyIndex === -1) {
+      const promiseJobs = await generateKeyPairPool(mnemonic, privateKey, -1, 0, 2);
```

Comment:
`-1` for no index?
`0` connections count?
`2` only 2 keys to generate?

Reply:
seems like an edge case scenario, maybe you could add a comment why these values are chosen?

Reply:
-1 would be no keys generated yet at all, 0 for connections number in order not to overgenerate those 2 ( countToGenerate + connectionsCount in the function algorithm otherwise)

Reply:
thanks! 👍  I'm sorry, can you add this comment to JS? 🙏

Reply:
92969f7